### ISSUE TITLE
Commands import de-duplicates secondary sources with an exact-match Set (same class of bug as #2733)

### ISSUE THREAD

**Issue by dyoshikawa** [dyoshikawa/rulesync#2741]
## Summary

`CommandsProcessor.loadToolFiles()` de-duplicates commands coming from a secondary source (OpenCode's inline `command` block, or a shared root matched by basename) with an exact-match `Set<string>`:

```ts
const seen = new Set(toolCommands.flatMap((command) => keysOf(command)));
```

`src/features/commands/commands-processor.ts:911`

## Why it matters

This is the same class of bug as #2733, which was fixed for skills and subagents in PR #2738. Two spellings that differ only in case pass the check as distinct commands, but they are written back into one `.rulesync/commands/` tree, where `commit.md` and `Commit.md` are a single file on macOS and Windows. The copy written last wins, so the precedence the sources were scanned in is inverted rather than honored — and no warning is emitted, because the exact comparison never saw a duplicate.

## Suggested direction

Reuse `ClaimedIdentities` from `src/types/feature-processor.ts` (added in PR #2738), which folds case and Unicode normalization and reports the source that claimed a name, so the warning can tell a cross-source collision from a same-source one. The `keysOf()` basename-matching logic is specific to commands and should be preserved: each candidate key would be claimed through the shared helper rather than added to a plain `Set`.

Note the trade-off documented for the skills fix: the rule applies on every platform, so on a case-sensitive filesystem the lower-precedence command is dropped with a warning rather than imported under its own name.

This was split out of the review of #2733 / PR #2738.
